raven/src/main/java/com/getsentry/raven/connection/AbstractConnection.java

```diff
@@ -78,6 +86,9 @@ public final void send(Event event) {
             doSend(event);
             waitingTime = baseWaitingTime;
         } catch (ConnectionException e) {
+            for (EventSendFailureCallback eventSendFailureCallback : eventSendFailureCallbacks) {
+                eventSendFailureCallback.onFailure(event, e);
```

Comment:
Although I guess on the next line we (on `master`) we do this:

```
logger.warn("An exception due to the connection occurred, a lockdown will be initiated.", e);
```

I suppose we're depending on the logging frameworks preventing infinite loops here. As noted in #151 I couldn't recreate the deadlock in logback (and linked to the code that I believe handles it), I'm not sure about the other frameworks -- but it seems like we're already expecting them to handle it on `master`.